axes/handlers/database.py

```diff
@@ -119,58 +119,48 @@ def user_login_failed(
         # 2. database query: Calculate the current maximum failure number from the existing attempts
         failures_since_start = 1 + self.get_failures(request, credentials)
 
-        # 3. database query: Insert or update access records with the new failure data
-        try:
-            attempt = AccessAttempt.objects.get(
+        if not (settings.AXES_ONLY_USER_FAILURES and username is None):
+            # 3. database query: Insert or update access records with the new failure data
+            attempt, created = AccessAttempt.objects.get_or_create(
                 username=username,
                 ip_address=request.axes_ip_address,
                 user_agent=request.axes_user_agent,
+                defaults={"failures_since_start": failures_since_start}
             )
             # Update failed attempt information but do not touch the username, IP address, or user agent fields,
             # because attackers can request the site with multiple different configurations
             # in order to bypass the defense mechanisms that are used by the site.
 
-            log.warning(
-                "AXES: Repeated login failure by %s. Count = %d of %d. Updating existing record in the database.",
-                client_str,
-                attempt.failures_since_start,
-                get_failure_limit(request, credentials),
-            )
-
             separator = "\n---------\n"
 
             attempt.get_data = Concat("get_data", Value(separator + get_data))
             attempt.post_data = Concat("post_data", Value(separator + post_data))
             attempt.http_accept = request.axes_http_accept
             attempt.path_info = request.axes_path_info
-            attempt.failures_since_start += 1
+            if not created:
+                attempt.failures_since_start += 1
```

Comment:
The updates for other counters and metadata should probably be handled in the same logical branch, and `attempt.save()` should only be invoked for requests that exist previously.